addons/pos_loyalty/models/pos_order.py

```diff
@@ -90,15 +117,14 @@ def confirm_coupon_programs(self, coupon_data):
         coupon_new_id_map = {k: k for k in coupon_data.keys() if k > 0}
 
         # Create the coupons that were awarded by the order.
-        coupons_to_create = {k: v for k, v in coupon_data.items() if k < 0 and not v.get('giftCardId')}
+        coupons_to_create = {k: v for k, v in coupon_data.items() if k < 0 and not v.get('giftCardId') and not v.get('backend_gift_card_id')}
```

Comment:
```suggestion
        coupons_to_create = {k: v for k, v in coupon_data.items() if k < 0 and not v.get('giftCardId') and not v.get('existing_gift_card_id')}
```

replace `existing_gift_card_id` with `backend_gift_card_id`